src/XenCert/XenCertCommon.py

```diff
@@ -274,15 +274,25 @@ def printCommand(argvs):
             pass
         else:
             if option == '-i':
-                temp_argvs[option_index+1] = ':'.join(hidePassword(temp_argvs[option_index+1].split(':'), 2))
+                temp_argvs[option_index+1] = ':'.join(getNoPasswordCmds(temp_argvs[option_index+1].split(':'), 2))
             else:
                 temp_argvs[option_index+1] = HIDDEN_PASSWORD
 
     for argv in temp_argvs:
         PrintToLog(argv)
         PrintToLog(' ')
 
-def hidePassword(cmd, password_index=-3):
+def getNoPasswordCmds(cmd, password_index=-3):
     cmd_with_hidden_password = cmd[:]
     cmd_with_hidden_password[password_index] = HIDDEN_PASSWORD
     return cmd_with_hidden_password
+
+def getNoPasswordConfig(config, storage_type):
+    config_with_hidden_password = dict(config)
+    if storage_type == 'iscsi' and config.get('chappassword'):
```

Comment:
It's good to check storage_type indeed here. while I still suggest to remove storage_type, just need
if config.get('chappassword') is not None:
    config_with_hidden_password['chappassword'] = HIDDEN_PASSWORD
if config.get('password') is not None:
    config_with_hidden_password['password'] = HIDDEN_PASSWORD
It' will make simpler, also hidden empty "" password.